core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -546,8 +548,8 @@ public int regexpInstr(String value, String regex, int position,
       final String methodName = "REGEXP_INSTR";
       final Pattern pattern = validateRegexPattern(regex, methodName);
 
-      if (checkPosOccurrenceParamValues(position, occurrence, occurrencePosition, value,
-          methodName) || regex.isEmpty()) {
+      if (regex.isEmpty() || !validatePosOccurrenceParamValues(position, occurrence,
+          occurrencePosition, value, methodName)) {
```

Comment:
Put `||` on a new line.

(Multi-line expressions shouldn't cross 'branches' of the expression tree in the middle of a line.)

Reply:
Thanks for the feedback Julian, corrected that.